### PR TITLE
Easily replaceable power cells in machines (mainly dispensers related)

### PR DESCRIPTION
**Pull request by MatiAvanti**
## About The Pull Request

Just what the title says, being able to replace power cell in a machine without actually breaking it. 

## Why It's Good For The Game
 It feels bad that you have to disassemble whole machine apart to replace a power cell. Power cell should be considered an easily replaceable part, its replacement should be less complicated.
This will lessen needs for tools for other departments such as botany/chemistry/bar and make their life's easier.
Power cells could be replaced by clicking on a machine with a battery in hand with  its maintenance hatch open.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

https://user-images.githubusercontent.com/58709092/215908458-7b88b96b-652d-49e1-99b2-ebe1e1ed63cc.mp4

https://user-images.githubusercontent.com/58709092/215908777-21b6d966-b545-43c4-a085-7d0b2fc3d822.mp4

https://user-images.githubusercontent.com/58709092/215909336-823f490a-e183-4fc8-99b1-679170a3180f.mp4

</details>

## Changelog
:cl:
add: Added Power Cell swapping.
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
can we restrict this until after #8372

**Comment by BriggsIDP** [comment]
This is a really nice QOL change for maybe jobs. Thanks!

**Comment by MatiAvanti** [comment]
I honestly suggest version without the need to use screwdriver. Power cells in my opinion should be replaceable without "specialized equipment" and it should be relatively simple to perform without professional knowledge.

**Comment by WrillWasTaken** [comment]
> del: Removed Nanite restricted acccess (one feature per PR they say)

if this was part of your PR previously and not part of the game already then there's no need to mention it in the changelog.

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -535,6 +535,23 @@ Class Procs:
 		return FALSE
 	return TRUE
 
+// Power cell in hand replacement
+/obj/machinery/attackby(obj/item/C, mob/user)
+	if(istype(C, /obj/item/stock_parts/cell) && panel_open)
+		locate(/obj/item/circuitboard/machine) in component_parts
```
Comment (PowerfulBacon) [inline code/game/machinery/_machinery.dm]: You are locating a circuitboard here and then doing nothing with it. Locate returns a value. What was the intent of this code?

File: code/game/machinery/_machinery.dm [hunk 2]
```diff
@@ -535,6 +535,23 @@ Class Procs:
...
+		for(var/obj/item/P in component_parts)
+			if(istype(P,/obj/item/stock_parts/cell))
+				if(user.transferItemToLoc(C, src))
+					user.put_in_active_hand(P)
+					component_parts+=C
+					component_parts-=P
+					RefreshParts()
+					playsound(src, 'sound/surgery/taperecorder_close.ogg', 50, FALSE)
+					to_chat(user, "<span class='notice'>you replaced [capitalize(P.name)]  with [C.name] via power cell socket.</span>")
```
Comment (PowerfulBacon) [inline code/game/machinery/_machinery.dm]:
```suggestion
					to_chat(user, "<span class='notice'>You replace [P.name] with [C.name].</span>")
```
Should be in present tense, capitalisation is inconsistent and in the middle of the sentance, capitalisation is missed at the start of the sentence.

File: code/game/machinery/_machinery.dm [hunk 3]
```diff
@@ -535,6 +535,23 @@ Class Procs:
...
+					break
+	else
+		..()
```
Comment (PowerfulBacon) [inline code/game/machinery/_machinery.dm]:
```suggestion
					return
	..()
```

Comment (PowerfulBacon) [reply code/game/machinery/_machinery.dm]: Does attackby return?

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/game/machinery/_machinery.dm (hunk 1 above)
Comment (MatiAvanti) [reply code/game/machinery/_machinery.dm]: My bad, thought locate also modifies component_parts. Checked again and feature is still working.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**